src/openms/source/PROCESSING/CENTROIDING/PeakPickerIM.cpp

```diff
@@ -2,35 +2,183 @@
+      std::cerr << "75% percentile of ion mobility difference is determined to be... " << threshold << std::endl;
+
+      std::vector<double> small_mz_diffs;
+      for (double diff : mz_diffs)
+      {
+        if (diff <= threshold)
+        {
+          small_mz_diffs.push_back(diff);
+        }
+      }
+
+      if (small_mz_diffs.empty())
+      {
+        std::cerr << "Warning: No valid small m/z differences found. Using default sampling rate. Using fixed sampling rate of 0.001 1/k" << std::endl;
+        return 0.001; // Default fallback value
+      }
+
+      // Step 2: Compute mode (most common spacing)
+      std::map<double, int> freq_map;
+      for (double diff : small_mz_diffs)
+      {
+        freq_map[diff]++;
+      }
+
+      double mode_sampling_rate = small_mz_diffs.front(); // Default to first value
+      int max_count = 0;
+      for (const auto& [diff, count] : freq_map)
+      {
+        if (count > max_count)
+        {
+          mode_sampling_rate = diff;
+          max_count = count;
+        }
+      }
+
+      // Compute final sampling rate
+      double sampling_rate = mode_sampling_rate;
+      std::cout << "Computed sampling rate: " << sampling_rate << std::endl;
+
+      return sampling_rate;
+    }
+
+    PeakPickerIM::PeakPickerIM() :
+        parameters_(getDefaultParameters())
+    {
+    }
+
+    // Destructor
+    PeakPickerIM::~PeakPickerIM()
+    {
+    }
+
+    // Returns a Param object with the default settings for peak picking.
+    Param PeakPickerIM::getDefaultParameters() const
+    {
+      Param p;
+      p.setValue("signal_to_noise", 0.0, "Signal to noise threshold for peak picking");
+      p.setValue("spacing_difference_gap", 0.0, "The extension of a peak is stopped if the spacing between two subsequent data points exceeds 'spacing_difference_gap * min_spacing'. 'min_spacing' is the smaller of the two spacings from the peak apex to its two neighboring points. '0' to disable the constraint. Not applicable to chromatograms.");
+      p.setValue("spacing_difference", 0.0, "Maximum allowed difference between points during peak extension, in multiples of the minimal difference between the peak apex and its two neighboring points. If this difference is exceeded a missing point is assumed (see parameter 'missing'). A higher value implies a less stringent peak definition, since individual signals within the peak are allowed to be further apart. '0' to disable the constraint. Not applicable to chromatograms.");
+      p.setValue("missing", 0, "Maximum number of missing points allowed when extending a peak to the left or to the right. A missing data point occurs if the spacing between two subsequent data points exceeds 'spacing_difference * min_spacing'. 'min_spacing' is the smaller of the two spacings from the peak apex to its two neighboring points. Not applicable to chromatograms.");
+      p.setValue("signal_to_noise", 0.0, "Signal to noise threshold for peak picking");
+      return p;
```

Comment:
these are params for the PPHiRes. You can simply create them within the pick() function locally?! Or create them once in the constructor and store them as private member?